src/modules/core/components/ClipboardCopy/ClipboardCopy.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState, useCallback } from 'react';
+import { defineMessages, MessageDescriptor } from 'react-intl';
+import copyToClipboard from 'copy-to-clipboard';
+
+import Button from '../Button';
+
+interface Props {
+  text?: MessageDescriptor;
+  value: string;
+}
+
+const MSG = defineMessages({
+  copyLabel: {
+    id: 'ClipboardCopy.copyLabel',
+    defaultMessage: `{valueIsCopied, select,
+      true {Copied}
+      false {Copy}
+    }`,
+  },
+});
+
+const displayName = 'ClipboardCopy';
+
+const ClipboardCopy = ({ value, text = MSG.copyLabel }: Props) => {
+  const [valueIsCopied, setValueIsCopied] = useState(false);
+  const handleClipboardCopy = useCallback(() => {
+    setValueIsCopied(true);
+    copyToClipboard(value);
+    const timer = setTimeout(() => setValueIsCopied(false), 2000);
+    return () => clearTimeout(timer);
+  }, [value, setValueIsCopied]);
```

Comment:
Nice call migrating to hooks! Unfortunately, I don't think `useCallback` calls the returned function when the component unmounts like `useEffect` does. I think in a case like this, it's best to use `useRef` in combination with `useEffect` to handle the timeout cleanup properly.